public/src/utils.js

```diff
@@ -351,7 +351,7 @@
 			if (!str) {
 				return '';
 			}
-			str = str.replace(utils.trimRegex, '');
+			str = ('' + str).replace(utils.trimRegex, '');
```

Comment:
ok, so, the reasoning for this one here is the following:

1) in `Privileges.addUserToPrivilegeTable()` the `member` value sent via the socket request is the integer value `uid`, `member: ui.item.user.uid` 

https://github.com/NodeBB/NodeBB/blob/9b308e4b07f2ded770aefc6ffb1b35e5166642ef/public/src/admin/manage/privileges.js#L144-L150

so ... when `Categories.setPrivilege()` is called, notice the `group.exists(data.member)` is called
https://github.com/NodeBB/NodeBB/blob/9b308e4b07f2ded770aefc6ffb1b35e5166642ef/src/socket.io/admin/categories.js#L51-L58

`Group.exists()` internally uses `utils.slugify()` which expects a string only, when getting an `int` we get an error

https://github.com/NodeBB/NodeBB/blob/9b308e4b07f2ded770aefc6ffb1b35e5166642ef/src/groups/index.js#L186-L194

did this come up before, the way to reproduce this is to visit ACP > Manage > Privileges and add a user to a specific category.

Reply:
It's a bug caused by a recent change I made, I will take a look at that https://github.com/NodeBB/NodeBB/commit/3455c27a8d3fe59ddf5791559ba283ccdeeaed80

Reply:
Fixed in https://github.com/NodeBB/NodeBB/commit/441dd86d29812dc5e5167ac7eb415b5c8468bc4e